compiler/symbol.go

```diff
@@ -0,0 +1,107 @@
+package compiler
+
+// This file manages symbols, that is, functions and globals. It reads their
+// pragmas, determines the link name, etc.
+
+import (
+	"go/ast"
+	"go/token"
+	"go/types"
+	"strings"
+
+	"github.com/tinygo-org/tinygo/loader"
+	"golang.org/x/tools/go/ssa"
+	"tinygo.org/x/go-llvm"
+)
+
+// globalInfo contains some information about a specific global. By default,
+// linkName is equal to .RelString(nil) on a global and extern is false, but for
+// some symbols this is different (due to //go:extern for example).
+type globalInfo struct {
+	linkName string // go:extern
+	extern   bool   // go:extern
+}
+
+// loadASTComments loads comments on globals from the AST, for use later in the
+// program. In particular, they are required for //go:extern pragmas on globals.
+func (c *Compiler) loadASTComments(lprogram *loader.Program) {
```

Comment:
OMG this function... yes I know it is just in a new location, but this just brings it to our collective attention. :smile_cat:

Reply:
Yeah I know but it's the only way to support pragmas on globals. I would really like to get rid of it as it's rather dirty.